src/cript/api/api.py

```diff
@@ -484,6 +502,404 @@ def _check_initial_host_connection(self) -> None:
+        """
+        schema_file_path = f"src/cript/schemas/{node_type.lower()}_schema.json"  # Path to the schema file
+        try:
+            with open(schema_file_path, "r") as file:
+                schema = json.load(file)
+            return schema
+        except FileNotFoundError:
+            print(f"Schema file not found for node type: {node_type}")
+
+        except json.JSONDecodeError:
+            print(f"Invalid JSON format in schema file for node type: {node_type}")
+
+    @staticmethod
+    def remove_keys_from_dict(
+        dictionary,
+        keys_to_remove=["uuid", "experiment_count", "inventory_count", "public", "created_at", "email", "model_version", "orcid", "uid", "updated_at", "username", "admin_count", "collection_count", "locked", "material_count", "member_count"],
+    ):
+        """Recursively remove keys from a dictionary."""
+        if not isinstance(dictionary, dict):
+            return dictionary
+        for key in keys_to_remove:
+            dictionary.pop(key, None)
+        for key, value in list(dictionary.items()):  # Use list() to avoid RuntimeError during iteration
+            if isinstance(value, dict):
+                API.remove_keys_from_dict(value, keys_to_remove)
+            elif isinstance(value, list):
+                for item in value:
+                    API.remove_keys_from_dict(item, keys_to_remove)
+        return dictionary
+
+    @staticmethod
+    def remove_duplicates(node_objects, duplicate_list):
+        removed_items = []  # To store removed materials
+        remaining_items = []  # To store remaining materials after removal
+
+        for item in node_objects:
+            if item["name"] in duplicate_list:
+                removed_items.append(item)
+            else:
+                remaining_items.append(item)
+
+        return remaining_items
+
+    @staticmethod
+    def send_patch_request_for_node(parent_node=None, payload=None):  # (self, base_url, entity_uuid, data, auth_token=None):
+        """
+        Sends a PATCH request to update an entity.
+        """
+        base_url = Config.host
+        parent_uuid = parent_node.uuid
+        parent_node_type = parent_node.node[0].lower()
+        url = f"{base_url}/api/v1/{parent_node_type}/{parent_uuid}"  # if entity_uuid else base_url
+        headers = {
+            "Content-Type": "application/json",
+        }
+        headers["Authorization"] = f"Bearer {Config.api_token}"
+        data = payload
+        response = requests.patch(url, json=data, headers=headers)
+        if response.status_code == 200:
+            print("\nEntity updated successfully.", response.status_code)
+        else:
+            print(f"Entity patch updated Failed. Status code: {response.status_code}, Response: {response.text}")
+        return response
+
+    @staticmethod
+    def send_delete_request_for_node(parent_node=None, payload=None):  # (self, base_url, entity_uuid, data, auth_token=None):
```

Comment:
should be `delete` function of API.

Also not a static function.

Reply:
same here - implement capsule